// svelte-preprocess/dist/processors/scss.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sass = exports.scss = void 0;
const tagInfo_1 = require("../modules/tagInfo");
const utils_1 = require("../modules/utils");
const prepareContent_1 = require("../modules/prepareContent");
const scss = (options) => ({
    async style(svelteFile) {
        const { transformer } = await import('../transformers/scss.js');
        let { content, filename, attributes, lang, alias, dependencies } = await (0, tagInfo_1.getTagInfo)(svelteFile);
        if (alias === 'sass') {
            options = {
                ...options,
                stripIndent: true,
                indentedSyntax: true,
            };
        }
        if (lang !== 'scss') {
            return { code: content };
        }
        content = (0, prepareContent_1.prepareContent)({ options, content });
        const transformed = await transformer({
            content,
            filename,
            attributes,
            options,
        });
        return {
            ...transformed,
            dependencies: (0, utils_1.concat)(dependencies, transformed.dependencies),
        };
    },
});
exports.scss = scss;
exports.sass = scss;
// both for backwards compat with old svelte-preprocess versions
// (was only default export once, now is named export because of transpilation causing node not to detect the named exports of 'svelte-preprocess' otherwise)
exports.default = scss;
